utils/docker_utils.go

```diff
@@ -0,0 +1,153 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"strings"
+	"time"
+)
+
+type ContainerConfig struct {
+	Name   string
+	Image  string
+	Mounts []mount.Mount
+}
+
+func NewDockerClient(host string) *client.Client {
+	if err := os.Setenv("DOCKER_HOST", host); err != nil {
+		log.Print(err)
+	}
+	cli, _ := client.NewEnvClient()
+	return cli
+}
+
+func ListContainer(cli *client.Client) {
+	ctns, err := cli.ContainerList(context.Background(), types.ContainerListOptions{All: true})
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, container := range ctns {
+		fmt.Printf("%s %s\n", container.ID[:10], container.Image)
+	}
+}
+
+func CreateContainer(cli *client.Client, config ContainerConfig) (string, error) {
+	log.Printf("Creating Container [%s]", config.Name)
+	ctx := context.Background()
+	hostConfig := container.HostConfig{
+		Mounts: config.Mounts,
+	}
+
+	dconfig := container.Config{
+		Image: config.Image,
+		Tty:   true,
+	}
+
+	b, err := cli.ContainerCreate(ctx, &dconfig, &hostConfig, nil, config.Name)
+
+	return b.ID, err
+}
+
+func StartContainer(cli *client.Client, id string) error {
+	log.Printf("Starting Container [%s]", id)
+	return cli.ContainerStart(context.Background(), id, types.ContainerStartOptions{})
+}
+
+func StopContainer(cli *client.Client, id string) error {
+	log.Printf("Stopping Container [%s]", id)
+	var timeout = 5 * time.Second
+	return cli.ContainerStop(context.Background(), id, &timeout)
+}
+
+func RemoveContainer(cli *client.Client, id string) error {
+	log.Printf("Removing Container [%s]", id)
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	return cli.ContainerRemove(ctx, id, types.ContainerRemoveOptions{})
+}
+
+func Write(cli *client.Client, id string, content []string, dest string) error {
```

Comment:
what does this write function do?

Reply:
Iterates over the content and write each one to the destination file